client/my-sites/domains/domain-management/settings/cards/name-servers-card.tsx

```diff
@@ -85,11 +85,23 @@ const NameServersCard = ( {
 			return false;
 		}
 
-		return nameservers.every( ( nameserver ) => {
+		return nameservers.some( ( nameserver ) => {
 			return ! nameserver || CLOUDFLARE_NAMESERVERS_REGEX.test( nameserver );
 		} );
 	};
 
+	const onlyWpcomNameservers = () => {
+		if ( ! nameservers || nameservers.length === 0 ) {
+			return false;
+		}
+
+		return (
+			nameservers.every( ( nameserver ) => {
+				return WPCOM_DEFAULT_NAMESERVERS_REGEX.test( nameserver );
+			} ) && nameservers.length === WPCOM_DEFAULT_NAMESERVERS.length
```

Comment:
`&& nameservers.length === WPCOM_DEFAULT_NAMESERVERS.length` this extra check means that `onlyWpcomNamservers` would return `false` if someone only added `ns1.wp.com` and `ns2.wp.com`.

Is that intended? If so, would `hasDefaultWpcomNameservers` be a better name for the method?

Reply:
The two checks needed to show/hide elements correctly 
- Is editing and entered any number of wpcom name servers, save is disabled, can only cancel or reset to the default (hasWpcomNameservers = true = save disabled)
- Is not editing and already has the default wpcomNameservers (onlyWpcomNameservers = true =hide the list of nameservers, ns1+2+3 are implied by "We are managing your domain") **So yeah with that in mind, hasDefaultWpcomNameservers is a superior name. Thanks!**